### PR TITLE
DB-7931: change export default compression format to bz2

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh5.12.2

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperationIT.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 package com.splicemachine.derby.impl.sql.execute.operations.export;
 
+import org.apache.commons.compress.compressors.bzip2.BZip2CompressorInputStream;
```
Comment (MurrayBrown) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperationIT.java]: surely, if we are adding this, we can remove the GZipInputStream on line 38?

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by genedavis** [comment]
I second the motion of making it an option.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperation.java [hunk 2]
```diff
@@ -208,7 +208,15 @@ public void writeExternal(ObjectOutput out) throws IOException {
             case "parquet": {
                 OperationContext<?> writeContext = dsp.createOperationContext(this);
                 long start = System.currentTimeMillis();
-                dataset.writeParquetFile(dsp, new int[]{}, exportParams.getDirectory(), exportParams.isCompression() ? "snappy" : "none", writeContext);
+                String compression = null;
+                if (exportParams.getCompression() == ExportFile.COMPRESSION.SNAPPY) {
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperation.java]: We should handle gz and bzip2 here too

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportParams.java [hunk 3]
```diff
@@ -117,8 +117,27 @@ private void setFormat(String format) throws StandardException {
         this.format = format;
     }
 
-    private void setCompression(Boolean compression) throws StandardException {
-        this.compression = compression;
+    private void setCompression(String compression) throws StandardException {
```
Comment (dgomezferro) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportParams.java]: This breaks the current uses of Export(), we should handle true/false too

File: db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ResultSetFactory.java [hunk 4]
```diff
@@ -1780,7 +1780,7 @@ NoPutResultSet getExportResultSet(NoPutResultSet source,
 									  Activation activation,
 									  int resultSetNumber,
 									  String exportPath,
-									  boolean compression,
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/iapi/sql/execute/ResultSetFactory.java]: Please make this change for Binary_export too

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportParams.java (hunk 3 above)
Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportParams.java]: We cannot keep the current export() API and introduce different compression options while still keeping true/false.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperation.java (hunk 2 above)
Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/export/ExportOperation.java]: Parquet does not support bzip2 compression

**Comment by jyuanca** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by dgomezferro** [comment]
Looks good to me! Thanks for the updates +1

**Pull request merged**